llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2524,10 +2529,15 @@ void VPReductionRecipe::execute(VPTransformState &State) {
     NewRed = createSimpleReduction(State.Builder, NewVecOp, Kind);
     if (RecurrenceDescriptor::isMinMaxRecurrenceKind(Kind))
       NextInChain = createMinMaxOp(State.Builder, Kind, NewRed, PrevInChain);
-    else
+    else {
+      // Sub reductions aren't commutative so the operands need to be swapped.
+      bool IsSub = Kind == RecurKind::Sub;
+      Value *LHS = IsSub ? PrevInChain : NewRed;
```

Comment:
For commutative operations the order doesn't matter, so I think you can always swap the order of PrevInChain/NewRed, avoiding the need for `IsSub`, `LHS` and `RHS`. That is best done in a separate (NFCI) patch though, because various tests require updating.